gen2-rgbd-projection/main.py

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env python3
+
+import argparse
+import cv2
+import depthai as dai
+import numpy as np
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--align-depth', dest='align_depth', action='store_true')
+parser.add_argument('--no-pcl', dest='no_pcl', action='store_true')
+parser.set_defaults(align_depth=False, no_pcl=False)
+
+args = parser.parse_args()
+align_depth = args.align_depth
+no_pcl = args.no_pcl
+
+if not no_pcl:
+    try:
+        from projector_3d import PointCloudVisualizer
+    except ImportError as e:
+        raise ImportError(
+            f"\033[1;5;31mError occured when importing PCL projector: {e}")
+
+############################################################################
+# USER CONFIGURABLE PARAMETERS (also see configureDepthPostProcessing())
+
+# parameters to speed up visualization
+stride = 4  # skip points in the depth image when projecting to 3d pointcloud, only matters if align_depth == False
+downsample_pcl = True  # downsample the pointcloud before operating on it and visualizing
+
+# StereoDepth config options.
+# whether or not to align the depth image on host (As opposed to on device), only matters if align_depth = True
+align_on_host = False
+lrcheck = True  # Better handling for occlusions
+extended = False  # Closer-in minimum depth, disparity range is doubled
+subpixel = True  # True  # Better accuracy for longer distance, fractional disparity 32-levels
+LRcheckthresh = 5
+confidenceThreshold = 200
+min_depth = 400  # mm
+max_depth = 20000  # mm
+speckle_range = 60
+############################################################################
+
+if (not lrcheck and not align_on_host) and align_depth:
+    print("Warning: Cannot align on device without lr check!")
+    align_on_host = True
+
+# Select camera resolution (oak-d-lite only supports THE_480_P  and THE_400_P depth)
+# res = {"height": 400, "width": 640,
+#        "THE_P": dai.MonoCameraProperties.SensorResolution.THE_400_P}
+res = {"height": 480, "width": 640,
```

Comment:
any specific reason why we are assigning only OAK-D-Lite Res ?

Reply:
That is my bad. I have an oak-d-lite, and I had hard-coded the desired resolution. The latest commit fixes this by still allowing the user to set the resolution, but now I get the height/width from the rgb and depth nodes to avoid hard-coding.

Reply:
https://github.com/luxonis/depthai-experiments/blob/1d9fe1084ba613824630efdb165dbc0b3a00e69b/gen2-rgbd-projection/main.py#L129